Added more json fields to AlertStatus webhook notification

MichaelUray opened the pull request:
**Description:**
Added more json fields to the AlertStatus webhook notification.
Added `make test` in documentation "Contributing to GoAlert".

MichaelUray committed:
Added more json fields to webhook notifications
for AlertStatus updates:

	Summary     string
	Details     string
	ServiceID   string
	ServiceName string
	Meta        map[string]string

Before these fields were only available in Alert notifications and it was not possible to send them to the user with acknowledge or close events.

MichaelUray committed:
Update contribution guidelines for testing command `make test`

MichaelUray committed:
Added PublicURL field to json for webhook
notification for Alart and AlartStatus
Before was that field not available and it was not possible to send them to the user with messages.

MichaelUray committed:
AlertBundle Webhook sends now PublicURL as well.

mastercactapus commented on notification/webhook/sender.go:
```diff
@@ -97,6 +107,7 @@ var _ nfydest.MessageSender = &Sender{}
 func (s *Sender) SendMessage(ctx context.Context, msg notification.Message) (*notification.SentMessage, error) {
 	cfg := config.FromContext(ctx)
 	var payload interface{}
+	pubURL := strings.TrimRight(cfg.General.PublicURL, "/")
```
Use `cfg.PublicURL()` (or `cfg.CallbackURL(...)`) as `cfg.General.PublicURL` is deprecated.

```suggestion
	pubURL := cfg.PublicURL()
```

MichaelUray committed:
cfg.PublicURL() gets used as cfg.General.PublicURL is deprecated.

Co-authored-by: Nathaniel Caza <mastercactapus@gmail.com>

MichaelUray committed:
Changed PublicURL to AlertURL
AlertURL contains the full link to the alert page, or to the alarm list of the according service if it is an alert bundle.

MichaelUray committed:
Merge branch 'target:master' into master

MichaelUray committed:
Replaced function calls for URL generation
fmt.Sprintf("%s/alerts/%d", pubURL, m.AlertID)
with
cfg.CallbackURL(fmt.Sprintf("/alerts/%d", m.AlertID))

MichaelUray committed:
Address review feedback: rename AlertURL to GoAlertURL and clean up

- Rename AlertURL to GoAlertURL as suggested, since the URL points to
  GoAlert itself, not to anything notification-specific
- Remove unused pubURL variable (cfg.CallbackURL is used instead)
- Remove unused strings import

MichaelUray commented:
@mastercactapus thanks for the review. All feedback is addressed in the latest commits:

1. **deprecated `cfg.General.PublicURL`** → now using `cfg.CallbackURL(fmt.Sprintf(...))` everywhere so short URL generation and any other callback behaviour is preserved ([sender.go](https://github.com/target/goalert/pull/4445/files#diff-b2a2f5d4c36e1e3f8a64eef3e1d79aab)).
2. **field naming** → renamed the field to `GoAlertURL` as suggested, so it is clear the URL points to GoAlert itself rather than any notification service.
3. **removed unused imports/variables** → the intermediate `pubURL` variable and the unused `strings` import are gone.

The final layout:
- **Alert** → `/alerts/<id>`
- **AlertBundle** → `/services/<id>/alerts`
- **AlertStatus** → `/alerts/<id>`

Ready for another look when you have a minute.

MichaelUray commented:
@mastercactapus friendly ping — your December review feedback was addressed in commits from early April (`cfg.CallbackURL` now used everywhere, field renamed to `GoAlertURL`, unused imports removed). The PR has been sitting at "ready for another look" since then. Anything else you'd like changed before this can move forward?
